clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -760,8 +760,8 @@ synchronized void maybeResolveSequences() {
                         // For the transactional producer, we bump the epoch if possible, otherwise we transition to a fatal error
                         String unackedMessagesErr = "The client hasn't received acknowledgment for some previously " +
                                 "sent messages and can no longer retry them. ";
-                        if (canBumpEpoch()) {
-                            epochBumpRequired = true;
+                        if (canTriggerEpochBump()) {
```

Comment:
Looks like here we'll go into the fatal error state if transaction protocol v2 is used.